app/routes/__index/domains/index.tsx

```diff
@@ -16,13 +22,51 @@ export const loader = async ({ request }: LoaderArgs) => {
 };
 
 export const action = async ({ request }: ActionArgs) => {
-  if (request.method === 'PUT') {
-    // TODO: Call the appropriate code to prolong record
-  } else if (request.method === 'DELETE') {
-    // TODO: Call the appropriate code to delete record
+  const username = await requireUsername(request);
+  const formData = await request.formData();
+  const intent = formData.get('intent');
+  const id = formData.get('id');
+  invariant(intent, 'Missing `intent` in posted form data');
+  invariant(id, 'Missing `id` in posted form data');
+
+  const recordID = Number(id);
+  const record = await getRecordById(recordID);
+
+  if (!record) {
+    throw new Response('The record is not found', {
+      status: 422,
```

Comment:
Why is this not `404`?  The client has requested a resource with a given `id`, and the database doesn't have it.  Isn't that the definition of `404`?

Reply:
Mario had this comment on this, which seemed to make sense for me:

@[Ririio](https://github.com/Ririio) Ririio [yesterday](https://github.com/DevelopingSpace/starchart/pull/324#discussion_r1133184112)
Question: In regards to this condition here, does this throw a status 404 if a specific record is not found? If that's the case you might want to throw a status [422](https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/422) instead to be more accurate